### PR TITLE
feature(melange): virtual libs

### LINKED ISSUE #6418: melange: virtual libs
Melange does not support virtual libs yet. There is a test called [virtual_lib](https://github.com/ocaml/dune/blob/528f0ed121b1ffa82eb63077bfde5603f03bd366/test/blackbox-tests/test-cases/melange/virtual_lib.t/run.t) that shows the error that happens when trying.

### PR DESCRIPTION
**Pull request by jchavarri**
fixes #6418.

### todo

- [x] remove hard-coded value in `Dep_rules` and pick right cm_kind based on mode
- [x] remove hard-coded value in `Vimpl.make`

### CODE REVIEW

File: test/blackbox-tests/test-cases/melange/virtual_lib.t/dune [hunk 1]
```diff
@@ -1 +1 @@
-(melange.emit (target output) (entries c) (libraries impl) (module_system es6))
+(melange.emit (target output) (entries c) (libraries impl) (module_system commonjs))
```
Comment (rgrinberg) [inline test/blackbox-tests/test-cases/melange/virtual_lib.t/dune:1]: Multiple lines would be appreciated.

File: src/dune_rules/dep_rules.ml [hunk 2]
```diff
@@ -10,13 +10,13 @@ let ooi_deps { vimpl; sctx; dir; obj_dir; modules = _; stdlib = _; sandbox = _ }
...
     match ml_kind with
-    | Intf -> Cm_kind.Cmi
+    | Intf -> (* todo : fix *) Lib_mode.Cm_kind.Melange Cmi
     | Impl -> vimpl |> Option.value_exn |> Vimpl.impl_cm_kind
   in
   let write, read =
     let ctx = Super_context.context sctx in
     let unit =
-      Obj_dir.Module.cm_file_exn obj_dir m ~kind:(Ocaml cm_kind) |> Path.build
+      Obj_dir.Module.cm_file_exn obj_dir m ~kind:cm_kind |> Path.build
```
Comment (rgrinberg) [inline src/dune_rules/dep_rules.ml:19]: Note that the ocamlobjinfo path is only needed for implementing a virtual library that is installed. We don't yet have the installation of melange libraries, so we won't be able to fix this yet.

Comment (jchavarri) [reply src/dune_rules/dep_rules.ml]: Thanks! I started tweaking code from here, then fixed the objs part, but I wasn't sure this needed changing.

**Comment by jchavarri** [comment]
@rgrinberg so this seems to be working —at least tests are passing locally— but I'm skeptical 😄 do you think there are more scenarios that I should test?

**Comment by rgrinberg** [comment]
It looks good for me. I'd say you should try using it internally now.

**Pull request merged**